tests/api/v2/datadog/scenarios_test.go

```diff
@@ -0,0 +1,193 @@
+package test
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/DataDog/datadog-api-client-go/api/v2/datadog"
+	"github.com/DataDog/datadog-api-client-go/tests"
+	"github.com/go-bdd/gobdd"
+)
+
+func TestScenarios(t *testing.T) {
+	s := gobdd.NewSuite(
+		t,
+		gobdd.WithIgnoredTags([]string{"@todo"}),
+		gobdd.WithBeforeScenario(func(ctx gobdd.Context) {
+			client := datadog.NewAPIClient(NewConfiguration())
+			tests.SetCtx(ctx, context.WithValue(context.WithValue(
+				context.Background(),
+				datadog.ContextAPIKeys,
+				map[string]datadog.APIKey{},
+			), clientKey, client))
+			tests.SetData(ctx, make(map[string]interface{}))
+			tests.SetCleanup(ctx, make(map[string]func()))
+		}), gobdd.WithAfterScenario(func(ctx gobdd.Context) {
+			tests.RunCleanup(ctx)
+		}),
+	)
+	tests.ConfigureSteps(s)
+
+	s.AddStep(`a valid "apiKeyAuth" key in the system`, aValidAPIKeyAuth)
+	s.AddStep(`a valid "appKeyAuth" key in the system`, aValidAppKeyAuth)
+	s.AddStep(`an instance of "([^"]+)" API`, anInstanceOf)
+	s.AddStep(`there is a valid user in the system`, user)
+	s.AddStep(`there is a valid role in the system`, role)
+	s.AddStep(`the user has the role`, userHasRole)
+	s.AddStep(`there is a valid permission in the system`, permission)
+	s.AddStep(`the permission is granted to the role`, permissionIsGrantedRole)
+	s.Run()
+}
+
+func aValidAPIKeyAuth(t gobdd.StepTest, ctx gobdd.Context) {
```

Comment:
Since v1 and v2 have the same auth pattern, can we common this out to a shared place and override later if they diverge?

Reply:
We can't due to types which are different depending on API version.